src/libraries/System.Private.CoreLib/src/System/IO/Directory.cs

```diff
@@ -34,6 +35,18 @@ public static DirectoryInfo CreateDirectory(string path)
             return new DirectoryInfo(path, fullPath, isNormalized: true);
         }
 
+        [UnsupportedOSPlatform("windows")]
+        public static DirectoryInfo CreateDirectory(string path, UnixFileMode unixCreateMode)
```

Comment:
Is this implementation ok for Windows where FileSystem is responsible for throwing, or should I create a separate implementation that immediately throws?
In case of the latter, I'll have separate `CreateDirectory` methods for Unix and Windows. How should I add the `UnsupportedOSPlatformAttribute`?

Reply:
>  I'll have separate CreateDirectory methods for Unix and Windows. How should I add the UnsupportedOSPlatformAttribute?

When you have a public API that is contained in multiple files, you would need to apply the attribute to all files. The "good" thing is that the API compat should catch if you miss it on one of the files.

For example:

https://github.com/dotnet/runtime/blob/655bc1d10b7d0c4626c78f2782c5d0b73044d9d9/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Linux.cs#L21-L24

https://github.com/dotnet/runtime/blob/655bc1d10b7d0c4626c78f2782c5d0b73044d9d9/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs#L26-L29

https://github.com/dotnet/runtime/blob/655bc1d10b7d0c4626c78f2782c5d0b73044d9d9/src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.iOS.cs#L24-L27

Reply:
> Is this implementation ok for Windows where FileSystem is responsible for throwing, or should I create a separate implementation that immediately throws?

Others should probably weigh in as well, but my initial thinking is that having code that does argument validation and other calls on Windows is a bit wasteful. So I guess if I was writing the code, I would split it and throw PNSE immediately.

Reply:
I've changed it to throw PNSE immediately.